lib/ansible/module_utils/vmware.py

```diff
@@ -912,13 +912,14 @@ def get_vm(self):
         vm_obj = None
         user_desired_path = None
         use_instance_uuid = self.params.get('use_instance_uuid') or False
-        if self.params['uuid'] and not use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
-        elif self.params['uuid'] and use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content,
-                                   vm_id=self.params['uuid'],
-                                   vm_id_type="instance_uuid")
-        elif self.params['name']:
+        if 'uuid' in self.params:
+            if self.params['uuid'] and not use_instance_uuid:
+                vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
+            elif self.params['uuid'] and use_instance_uuid:
+                vm_obj = find_vm_by_id(self.content,
+                                       vm_id=self.params['uuid'],
+                                       vm_id_type="instance_uuid")
+        if self.params['name']:
```

Comment:
```suggestion
        if 'name' in self.params:
```

Reply:
Hi @Akasurde , Thanks for the comment.
Here I assume that atleast 'name' is basic parameter to be passed to get the VM. Where as moid and uuid are not common paramter which require check.
Please let me know if you think otherwise
Thanks

Reply:
Yes, `name` is basic parameter, but since we are `here` we can fix this as well.